### ISSUE TITLE
[Feature] Flexible interval for LoggerHook

### ISSUE THREAD

**Issue by JunweiZheng93** [open-mmlab/mmengine#867]
### What is the feature?

Hi there,

I appreciate your effort for developing this amazing package. I found it is not very convenient to set the `interval` in `LoggerHook` when using multiple GPUs, since the length of the dataloader changes when using different number of GPUs and I want to log my results once an epoch in my case. In order to get the correct `interval` for logging once an epoch, I need to calculate the length of the dataloader according to the number of GPUs I am using.

Is there an easier way to set `interval` according to the number of GPUs I am using for the sake of logging results once an epoch? If not, is it possible to reuse `log_metric_by_epoch` and then set `interval=1` to achieve that functionality? I think other developers who prefer logging results once an epoch will also like this feature.

Have a nice day.

**Comment by HAOCHENYE** [comment]
Hi, thanks for your feedback :smile: , Actually, the **smoothing window** and logging frequency is decoupled, if you want to count the average loss of a epoch, you can set the `window_size=epoch` for [here](https://github.com/open-mmlab/mmengine/blob/f10b5cefd97ccc03bb78bc40b39fb41428b46ef9/mmengine/runner/runner.py#L160). 

But if you mean that you want to log the training loss only once each epoch, maybe you can just set the interval as larger as possible, and set `ignore_last=False`.

**Comment by JunweiZheng93** [comment]
Hi, thx for your suggestion. That's exactly what I need. Close.
